src/main.cpp

```diff
@@ -1021,19 +1036,23 @@ bool ContextualCheckZerocoinSpendNoSerialCheck(const CTransaction& tx, const Coi
         libzerocoin::SpendType expectedType = libzerocoin::SpendType::SPEND;
         if (tx.IsCoinStake())
             expectedType = libzerocoin::SpendType::STAKE;
-        if (spend.getSpendType() != expectedType) {
+        if (spend->getSpendType() != expectedType) {
             return error("%s: trying to spend zPIV without the correct spend type. txid=%s\n", __func__,
                          tx.GetHash().GetHex());
         }
     }
 
+    //Reject V1 old serials.
+    if (spend->getVersion() < libzerocoin::PrivateCoin::PUBKEY_VERSION) {
+        return error("%s : zPIV v1 serial spend not spendable\n", __func__,
+                     spend->getCoinSerialNumber().GetHex(), tx.GetHash().GetHex());
+    }
```

Comment:
This needs to be checked against block height to avoid locking initial sync. We have to accept v1 spends already in the blockchain.

Reply:
Also ``return error("%s : zPIV v1 serial spend not spendable\n", __func__,               spend->getCoinSerialNumber().GetHex(), tx.GetHash().GetHex());`` need one more conversion specifier or one less arg.

Reply:
ok great, good catch 👌